src/app/services/baw-api/website-status/website-status.service.ts

```diff
@@ -0,0 +1,27 @@
+import { HttpClient } from "@angular/common/http";
+import { Inject, Injectable } from "@angular/core";
+import { ApiShow } from "@baw-api/api-common";
+import { WebsiteStatus } from "@models/WebsiteStatus";
+import { API_ROOT } from "@services/config/config.tokens";
+import { Observable, catchError, map, of } from "rxjs";
+
+@Injectable()
+export class WebsiteStatusService implements ApiShow<WebsiteStatus> {
+  public constructor(
+    private http: HttpClient,
+    @Inject(API_ROOT) private apiRoot: string
+  ) {}
+
+  // if the server is fully down, we will not receive a response from the API
+  // we can therefore not determine the status of the server and we should return "null"
+  public show(): Observable<WebsiteStatus | null> {
+    const endpoint = `${this.apiRoot}/status`;
+
+    return this.http
+      .get<WebsiteStatus>(endpoint)
+      .pipe(
+        map((response) => new WebsiteStatus(response)),
+        catchError(() => of(null))
```

Comment:
does the error have any information we're discarding here?

Reply:
We discard the [`BawApiError`](https://github.com/QutEcoacoustics/workbench-client/blob/13ad6e6347ee18b8160f95de046951fe4a265368/src/app/helpers/custom-errors/baw-api-error.ts#L7) information.

![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/464b21f0-4ddc-4e50-ab92-797b12f5db75)

Although, I just realized that we can probably just have a `noResponse` model that we use (on the `website-status.component.ts`) if we receive model that is `instanceof BawApiError`

Reply:
you can still return null; but the golden rule is don't squash errors. at a minimum log an error, better yet handle it somehow...